### PR TITLE
(TGCRP) - /tg/ catgirl refugee program

### LINKED ISSUE #9999: Humanitarian Crisis in TG station
**Problem Description**:
They're removing felinids

**What did you expect to happen**: 
They don't remove felinids

**What happened instead**:
There's a caravan of felinids heading our way

**Why is this bad/What are the consequences:**
We don't got no felinids in our codebase

**Steps to reproduce the problem**:
1. Remove felinids
2. Be listed server
3. Bawk at refugee crisis

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Gamemode was forum vote

### PR DESCRIPTION
**Pull request by tigercat2000**
Our /tg/ brotheren have been driven out, suffering, with the advent of tgstation/tgstation/pull/40978. In a show of solidarity, we, the Paradise staff, have decided to create a safe haven for all catgirls.

Please welcome these refugees with open arms and love in your heart.

 

Fixes #9999

:cl:
add: SSoverlays human mobs
add: Catgirls!
add: /tg/'s catgirl refugees
add: Ears are now organs
fix: The suffering of thousands of displaced kittens
/:cl:

### CODE REVIEW

**Comment by Mitchs98** [comment]
I never expected this to actually happen. I have no words. I am however laughing my ass off.

**Comment by KasparoVy** [comment]
DO IT

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -1527,8 +1526,9 @@
 	var/obj/item/organ/external/head/head_organ = get_organ("head")
 	var/datum/sprite_accessory/hair/hair_style = GLOB.hair_styles_full_list[head_organ.h_style]
 	var/icon/hair = new /icon("icon" = hair_style.icon, "icon_state" = "[hair_style.icon_state]_s")
-
-	return image(get_icon_difference(get_eyecon(), hair), layer = LIGHTING_LAYER + 1) //Cut the hair's pixels from the eyes icon so eyes covered by bangs stay hidden even while on a higher layer.
+	var/mutable_appearance/MA = mutable_appearance(get_icon_difference(get_eyecon(), hair), layer = LIGHTING_LAYER + 1)
```
Comment (KasparoVy) [inline code/modules/mob/living/carbon/human/human.dm]: Niiiiice.

**Comment by Necaladun** [comment]
:-1: 
 
When TG sends its catpeople, they're not sending their best. They're not sending you. They're not sending you. They're sending people that have lots of autism, and they're bringing that autism with us. They're bringing ERP. They're bringing cringe. They're metagamers. And some, I assume, are robust players.

File: code/modules/mob/living/carbon/human/species/human.dm [hunk 2]
```diff
@@ -17,4 +17,11 @@
...
-	//Has standard darksight of 2.
\ No newline at end of file
+	//Has standard darksight of 2.
+
+/datum/species/human/cat
+	name = "Felinid"
+	name_plural = "Felinids"
+	blurb = "CATGIRLS OMG"
+
+	mutantears = /obj/item/organ/internal/ears/cat
```
Comment (KasparoVy) [inline code/modules/mob/living/carbon/human/species/human.dm]: Would it now make sense to set up Tajaran & Vulpkanin mutantears and get 'em off the head sprites? This also means we may be able to have different selectable ears.

:thinking:

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 3]
```diff
@@ -105,68 +105,20 @@ Please contact me on #coderbus IRC. ~Carn x
 
 /mob/living/carbon/human
 	var/list/overlays_standing[TOTAL_LAYERS]
-	var/list/misc_effect_overlays = list() //Overlays that are applied at a custom layer (defined in each image's .layer property) outside of standard overlay application. Updated in update_misc_effects()
 	var/previous_damage_appearance // store what the body last looked like, so we only have to update it if something changed
 	var/icon/skeleton
 	var/list/cached_standing_overlays = list() // List of everything currently in a human's actual overlays
 
-//UPDATES OVERLAYS FROM OVERLAYS_LYING/OVERLAYS_STANDING
```
Comment (KasparoVy) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Holy shit RIP

File: code/__DEFINES/misc.dm [hunk 4]
```diff
@@ -204,40 +204,44 @@
...
+#define FACEMASK_LAYER			11
+#define HEAD_LAYER				10
+#define COLLAR_LAYER			9
+#define HANDCUFF_LAYER			8
+#define LEGCUFF_LAYER			7
+#define L_HAND_LAYER			6
+#define R_HAND_LAYER			5
+#define TARGETED_LAYER			4	//BS12: Layer for the target overlay from weapon targeting system
+#define FIRE_LAYER				3	//If you're on fire
+#define MISC_LAYER				2
```
Comment (KasparoVy) [inline code/__DEFINES/misc.dm]: Does this still render above lighting overlays?

**Comment by KasparoVy** [comment]
Just how much more efficient is mob icons now?

File: code/modules/mob/living/carbon/human/human.dm [hunk 5]
```diff
@@ -1527,8 +1526,9 @@
 	var/obj/item/organ/external/head/head_organ = get_organ("head")
 	var/datum/sprite_accessory/hair/hair_style = GLOB.hair_styles_full_list[head_organ.h_style]
 	var/icon/hair = new /icon("icon" = hair_style.icon, "icon_state" = "[hair_style.icon_state]_s")
-
-	return image(get_icon_difference(get_eyecon(), hair), layer = LIGHTING_LAYER + 1) //Cut the hair's pixels from the eyes icon so eyes covered by bangs stay hidden even while on a higher layer.
+	var/mutable_appearance/MA = mutable_appearance(get_icon_difference(get_eyecon(), hair), layer = LIGHTING_LAYER + 1)
+	MA.plane = LIGHTING_PLANE
```
Comment (KasparoVy) [inline code/modules/mob/living/carbon/human/human.dm]: YES MAN AHHAAAHAHAHAHA PLANESSSSSS YEEESSSSSS!

**Comment by DesolateG** [comment]
No

**Comment by Allfd** [comment]
NYAAAA~~~~~~

Test merge inbound!

**Comment by BakKeener** [comment]
this is quite a huge meme that frightens us and I feel like we are gonna go 200 pop suddenly because this gets added and see a lot of bans handed out but have some new players stay around, also this would be hilarious to watch unfold if I just take it on the lighter side.

**Comment by variableundefined** [comment]
We're abolishing the 24-hours rules.

**Comment by Xantholne** [comment]
Sure why not.

**Comment by Allfd** [comment]
NYAAA~~~~

**Comment by KasparoVy** [comment]
Don't revert wussies :sunglasses:

**Comment by Quantum-M** [comment]
But why?

**Comment by DarkPyrolord** [comment]
Nya~

**Comment by variableundefined** [comment]
Ok. Jokes over. The backend changes will be redone again.

**Comment by Necaladun** [comment]
LAUGHING TIME IS OVER

https://www.youtube.com/watch?v=ekxn5KLv9XQ

**Comment by KasparoVy** [comment]
Revert the revert

**Comment by BakKeener** [comment]
no, this actually made some of our players pretty mad and it was pretty bullshit from where I was standing and having to deal with it.

**Comment by TheRespectedMan** [comment]
Take in our refugees. What are you, racist? 
They're just like furries, except they also like anime! You will all be fine.

**Comment by KasparoVy** [comment]
tHe BeSt Of BoTh WoRlDs

**Comment by taukausanake** [comment]
F

**Pull request merged**